trio/_core/_local.py

```diff
@@ -28,14 +39,23 @@ class RunVar(metaclass=Final):
 
     """
 
-    _NO_DEFAULT = object()
-    _name = attr.ib()
-    _default = attr.ib(default=_NO_DEFAULT)
+    _NO_DEFAULT = NoValue()
+    _name: str = attr.ib()
+    _default: T | NoValue = attr.ib(default=_NO_DEFAULT)
+
+    @overload
+    def get(self, default: T) -> T:
+        ...
+
+    @overload
+    def get(self, default: NoValue = _NO_DEFAULT) -> T | NoValue:
+        ...
 
-    def get(self, default=_NO_DEFAULT):
+    def get(self, default: T | NoValue = _NO_DEFAULT) -> T | NoValue:
```

Comment:
If it can't return `NoValue` there's no point in doing the `@overload`, so removing those.

Not totally in love with using features whose PR's are yet to be merged, and not supported in pyright either. But pretty minor and seems likely to be supported soon, so might as well.